Update enable_i2c.markdown (#16073)
home-assistant/home-assistant.github.io: source/hassio/enable_i2c.markdown

  present at boot time.

The I2C devices should now be present under /dev.

### From Hassio Terminal

At the hassio main terminal, login as root, then type `login` to access the shell. The following enables i2c. 
```
mount /dev/sda1 /mnt
mkdir -p /mnt/CONFIG/modules
echo -ne i2c-dev>/mnt/CONFIG/modules/rpi-i2c.conf
echo dtparam=i2c_vc=on >> /mnt/CONFIG/config.txt
echo dtparam=i2c_arm=on >> /mnt/CONFIG/config.txt
sync
reboot

> ```

Reviewer: Please make sure the build passes. Check the build output for reasons why the build failed.

Additional, this section does not explain how the user can access that terminal.

Author: Sorry, I'm a n00b.  This is my first commit.  I'll figure it out later today.

Author: Updated the request.

  present at boot time.

The I2C devices should now be present under /dev.

> 
> ### From Home Assistant Operating System Terminal

Reviewer: I think this needs to be a level 2

```suggestion
## From Home Assistant Operating System Terminal
```

> At the Home Assistant Operating System Terminal, physical-access, main terminal, 
>   1. login as root
>   2. type `login` to access the shell. 
>   3. type the following to enable i2c. 

Reviewer: ```suggestion

Alternatively, by attaching a keyboard and screen to your device, you can access the physical terminal to the Home Assistant operating system.

You can enabled i2c via this terminal:

- Login as `root`.
- Type `login` and press enter to access the shell.
- Type the following to enable i2c:
```

> ```shell
> mount /dev/sda1 /mnt
> mkdir -p /mnt/CONFIG/modules
> echo -ne i2c-dev>/mnt/CONFIG/modules/rpi-i2c.conf
> echo dtparam=i2c_vc=on >> /mnt/CONFIG/config.txt
> echo dtparam=i2c_arm=on >> /mnt/CONFIG/config.txt
> sync
> reboot
> ```

Reviewer: Indent so it becomes part of the above list.

````suggestion
  ```shell
  mount /dev/sda1 /mnt
  mkdir -p /mnt/CONFIG/modules
  echo -ne i2c-dev>/mnt/CONFIG/modules/rpi-i2c.conf
  echo dtparam=i2c_vc=on >> /mnt/CONFIG/config.txt
  echo dtparam=i2c_arm=on >> /mnt/CONFIG/config.txt
  sync
  reboot
  ```
````

  present at boot time.

The I2C devices should now be present under /dev.

## From Home Assistant Operating System Terminal

> 
> Alternatively, by attaching a keyboard and screen to your device, you can access the physical terminal to the Home Assistant operating system.

Reviewer: ```suggestion
Alternatively, by attaching a keyboard and screen to your device, you can access the physical terminal to the Home Assistant Operating System.
```

You can enabled i2c via this terminal:

- Login as `root`.
- Type `login` and press enter to access the shell.
- Type the following to enable i2c:
  ```shell
  mount /dev/sda1 /mnt

>   mkdir -p /mnt/CONFIG/modules

Reviewer: This seems off, the mount is made on `/mnt`, next the modules folder is created in `/mnt/CONFIG`.

Additionally, is `/dev/sda1` guaranteed/fixed?

Author: We mount /dev/sda1, which is the boot partition to /mnt.  Then we recursively create the /mnt/CONFIG/modules folder.  Due to the -p option, it will also create the /mnt/CONFIG folder.  

Yes it is always /dev/sda1 or /dev/mmcblk0p1.  /dev/sda1 is your /mnt/boot on the primary operating system.   /dev/sda1  is also raspberry pi OS /boot partition, and /firmware/boot on Ubuntu for raspberry pi.  It is the FAT partition detected by windows when you plug in the drive to a computer. 

If you plug in an additional USB drive, that will be /dev/sdb1

I have an example in add-on format here  https://github.com/adamoutler/HassOSConfigurator/blob/main/Pi4EnableI2C/run.sh

Reviewer: Unresolved the conversation, as you just confirmed here that this is not stable.

Users will copy and paste this and thus it might not work.

Author: This was resolved via mention of other block devices   https://github.com/home-assistant/home-assistant.io/pull/16073/commits/68762a39dea58fb259265caff6cf9c86bf3faab3

> - Type the following to enable i2c, you may breplace sda1 with sdb1 or mmcblk0p1 depending on your platform:

Reviewer: ```suggestion
- Type the following to enable i2c, you may need to replace `sda1` with `sdb1` or `mmcblk0p1` depending on your platform:

```
